src/device/cuda/atomics.jl

```diff
@@ -319,6 +319,9 @@ macro atomic(ex)
     if ex.head == :(=)
         ref = ex.args[1]
         rhs = ex.args[2]
+        if typeof(rhs) <: Number || typeof(rhs) <: String || typeof(rhs) <: Symbol
```

Comment:
That's not what I meant, instead of checking against a blacklist of non-Expr types, it's easier to (whitelist) check for what's allowed: `Expr`.

Reply:
Ahh okay I'll fix this